model/job-management/src/main/proto/beam_expansion_api.proto

```diff
@@ -46,6 +46,9 @@ message ExpansionRequest {
   // A namespace (prefix) to use for the id of any newly created
   // components.
   string namespace = 3;
+
+  // The type of environment to use for the expanded transform.
+  org.apache.beam.model.pipeline.v1.Environment environment = 4;
```

Comment:
It'd be better to refer to environments already in (or placed in) the components above, if just to facilitate sharing with existing transforms.

Reply:
Completely agree, makes sense to add the environments to components.